independent-projects/tools/platform-descriptor-resolver-json/src/main/java/io/quarkus/platform/descriptor/resolver/json/QuarkusJsonPlatformDescriptorResolver.java

```diff
@@ -451,10 +451,11 @@ private QuarkusPlatformDescriptor loadPlatformDescriptor(ArtifactResolver mvn, f
         // check whether the quarkus-platform-descriptor-json used in the platform is already on the classpath
         final String pomPropsPath = "META-INF/maven/" + ToolsConstants.IO_QUARKUS + "/" + QUARKUS_PLATFORM_DESCRIPTOR_JSON
                 + "/pom.properties";
-        final InputStream is = getCpResourceAsStream(pomPropsPath);
-        if (is != null) {
+        final URL url = Thread.currentThread().getContextClassLoader().getResource(pomPropsPath);
+        // FIXME ia3andy temporary hack to work with jars
+        if (url != null && !url.toString().startsWith("jar:")) {
             final Properties props = new Properties();
```

Comment:
@aloubyansky ⬆️ this is my dirty hack until we find a nicer solution

Reply:
Why do you care where it is coming from?

Reply:
@aloubyansky TBH, I was just trying to make it work to continue coding the codestarts.

Before, it was using a `ClassPathResourceLoader` as soon as it was able to resolve the properties (as a stream), but this loader doesn't work with jars when getting `Path`.

Maybe we should just improve the `ClassPathResourceLoader` to make it a little smarter and support jars? Is this what you meant when pointing me to `ClassPathUtils`?

Reply:
I don't understand what this change is about, tbh. It will load the props whether they are in a JAR or a dir.

Reply:
@aloubyansky look bellow, when the properties is read successfully, it sets the classloader and, it then uses the `ClassPathResourceLoader`, only when the classloader is null that it will use the `ZipResourceLoader`.

We can just ignore my change here, this is just a temporary hack (it was failing without it).